signer/src/block_observer.rs

```diff
@@ -149,7 +150,11 @@ where
         known_deposit_requests: &DepositRequestMap,
         block: bitcoin::Block,
     ) -> Result<(), error::Error> {
-        let stacks_blocks = self.stacks_client.get_last_tenure_blocks().await?;
+        let block_id = StacksBlockId::first_mined();
+        let stacks_blocks = self
+            .stacks_client
+            .fetch_unknown_ancestors(block_id, &self.storage)
+            .await?;
```

Comment:
This needs to change. I didn't want to wade into this API for this PR, so I did the simplest thing without changing the spirit of the code already here.

Reply:
I don't really follow how this works either. If we get the unknown ancestors of the first mined block - wouldn't we miss all blocks that have been mined since the first mined block?

Reply:
Yeah I didn't want to make too many changes in this PR, but I know what we can do.

Reply:
Alright 👌